text/0000-action-menu.md

```diff
@@ -0,0 +1,214 @@
+
+5. **Interactions with other components**: How does the ActionMenu behave if other components on the page have overlapping z-indexes, or if there's a modal/dialog open? How does it behave when nested inside another container or component with special properties (like overflow hidden)?
+
+A ActionMenu component should be capable of handling a variety of content within its items, as well as a varying number of items.
+
+# Motivation
+
+The motivation for this feature is to provide a menu that is flexible in terms of styling, while also being accessible. It should integrate seamlessly with Pando's existing API, offer an easy-to-use interface for developers, and an intuitive and accessible experience for end users.
+
+# Detailed design
+
+The newly proposed `ActionMenu` embodies a headless component. This means it provides the functional logic of a menu without imposing any default styles. It is a composite of multiple sub-components, allowing it to adapt to and satisfy the requirements of any application. Moreover, it incorporates three paramount principles: accessibility (A11y), high performance, and ease of use.
+
+1. **Accessibility (A11y)**: Prioritizing accessibility, the `ActionMenu` is designed to ensure that all users, regardless of their abilities, can smoothly interact with it. Every component part will incorporate accessibility standards as per the Web Content Accessibility Guidelines (WCAG) to ensure the platform's inclusivity.
+
+2. **High Performance**: The `ActionMenu` is optimized for high performance. It's designed to be lightweight and responsive, ensuring fast loading times and a smooth user experience, irrespective of the application's scale.
+
+3. **Ease of Use**: While offering complex functionality, the `ActionMenu` is designed with user-friendliness in mind. Developers can easily combine its components to create a custom ActionMenu tailored to their application's needs.
+
+Here is a rundown of the `ActionMenu` structure:
+
+- **`ActionMenu`**: This component acts as a wrapper for all other components, maintaining the state of the menu. Its design provides a flexible user interface that can be adapted to suit the unique requirements of any application, while preserving optimal accessibility.
+
+- **`ActionMenu.Button`**: This component is the button that triggers the menu. Developers can define the appearance of the button, allowing for a high degree of visual customization while maintaining ease of interaction.
+
+- **`ActionMenu.Overlay`**: This component encapsulates the `ActionList` and is the actual menu displayed when the `ActionMenu.Button` is clicked. It is designed with user accessibility in mind, ensuring that the menu is easily navigable.
```

Comment:
Unnecessary component I would say (i.e. could create an abstraction to manage it internally in another).